src/ucp/wireup/select.c

```diff
@@ -233,10 +233,18 @@ ucp_wireup_check_keepalive(const ucp_wireup_select_params_t *select_params,
                            const char *title, const char **flag_descs,
                            char *reason, size_t max)
 {
-    ucp_worker_h worker = select_params->ep->worker;
+    ucp_worker_h worker  = select_params->ep->worker;
+    uint64_t iface_flags = UCT_IFACE_FLAG_EP_CHECK;
     char title_keepalive[128];
     char title_ep_check[128];
 
+    if (ucp_ep_init_flags_has_cm(select_params->ep_init_flags) &&
```

Comment:
@hoopoepg

Reply:
may be add evaluation of AM_BCOPY into ucp_ep_do_keepalive?
in this case we will be able to process KA via ep_check on shm based EP's

Reply:
makes sense
will be needed for CMA/KNEM/CUDA-IPC